examples/client/wayland_client.c

```diff
@@ -117,7 +117,7 @@ make_shm_pool(struct wl_shm* shm, int size, void **data)
 
     posix_fallocate(fd, 0, size);
 
-    *data = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
+    *data = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE, fd, 0);
```

Comment:
This is the wrong `mmap` call to change. Here, we're creating a mapping for submitting a SHM buffer - mapping this as `MAP_PRIVATE` ensures that the server will not be able to read anything we've written to the buffer :grin: 

[This](https://github.com/MirServer/mir/blob/2a06ae488d95b509470331716d55e5d10c54b8f9/src/platforms/wayland/displayclient.cpp#L662) is the site where we `mmap` a keymap sent from the server.